modules/gapi/src/streaming/onevpl/accelerators/surface/dx11_frame_adapter.cpp

```diff
@@ -114,6 +114,16 @@ MediaFrame::View VPLMediaFrameDX11Adapter::access(MediaFrame::Access mode) {
     }
 }
 
+cv::util::any VPLMediaFrameDX11Adapter::getHandle() {
+    auto surface_ptr_copy = get_surface();
+    Surface::data_t& data = surface_ptr_copy->get_data();
+    NativeHandleAdapter* native_handle_getter = reinterpret_cast<NativeHandleAdapter*>(data.MemId);
+
+    mfxHDLPair handle{};
+    native_handle_getter->get_handle(data.MemId, reinterpret_cast<mfxHDL&>(handle));
+    return handle;
```

Comment:
is not safe method in total for the moment

before using it, please make sure that Frame is not destroyed: somebody somewhere stores View on that (at least)
I had an issue with Frame  due to our lack synchronization, it was destroying  while invoked `blobParam`. For sadness, i don't remember use-case. But it cause too much pain

Reply:
additionally, please reuse this method in subsequent methods of this FrameAdapter. I could see similar code below

Reply:
Removed code duplication.
Regarding safety:  `View` hasn't been created at this point if I get it right. I don't see any artifacts/memory corruption. Please, let me know if you remember the use case with `blobParams` and how to fix it - didn't find any hints in existing code

Reply:
getHandle() must be not used as public interface, because it works outside of the alike reference counting mechanism protection. I mean 'handle' should not be remembered and cached separately from MediaFrame or View.
MediaFrame works with vpl surface inside, which is a part of cached pool. Surface (or handle in your case) is marked as 'busy' or 'free' sometimes and may change its state. So it must be strong guarantee that the following situation is forbidded

```
mfxHandle h;
{
MediaFrame frame = ...;
h = frame.getHandle();
} //reference counting becomes 0
// h - is invalid here
```

I faced some similar threading related situation in giebackend, but I don't remember details

Reply:
This method/adapter is not public. Added a comment to use with caution. For now leaving as is and closing the comment. Thanks for explanation